File: src/lib/types/artifact.ts
export interface Artifact {
	id: string;
	day_attempt_id: string;
	artifact_type: 'file' | 'link' | 'code' | 'note';
	title: string;
	content: string;
	file_path: string | null;
	file_size: number | null;
	mime_type: string | null;
	url: string | null;
	created_at: string;
}

export interface CreateArtifactInput {
	day_attempt_id: string;
	artifact_type: string;
	title: string;
	content: string;
	file_path?: string;
	file_size?: number;
	mime_type?: string;
	url?: string;
}

export interface UpdateArtifactInput {
	title?: string;
	content?: string;
}

File: src-tauri/src/db/models/artifact.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Artifact {
    pub id: String,
    pub day_attempt_id: String,
    pub artifact_type: String,
    pub title: String,
    pub content: String,
    pub file_path: Option<String>,
    pub file_size: Option<i32>,
    pub mime_type: Option<String>,
    pub url: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateArtifactInput {
    pub day_attempt_id: String,
    pub artifact_type: String,
    pub title: String,
    pub content: String,
    pub file_path: Option<String>,
    pub file_size: Option<i32>,
    pub mime_type: Option<String>,
    pub url: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateArtifactInput {
    pub title: Option<String>,
    pub content: Option<String>,
}
